src/BigPipeTrait.php

```diff
@@ -13,12 +15,30 @@
  */
 trait BigPipeTrait {
 
+  /**
+   * Flag for JS not supported by driver.
+   *
+   * @var bool
+   */
+  protected $bigPipeNoJS;
+
+  /**
+   * Skip Big Pipe BeforeStep.
+   *
+   * @var bool
+   */
+  protected bool $skipBigPipeBeforeStep = FALSE;
+
   /**
    * Prepares Big Pipe NOJS cookie if needed.
    *
    * @BeforeScenario
    */
   public function bigPipeBeforeScenarioInit(BeforeScenarioScope $scope) {
+    // Allow to skip resetting cookies on step.
+    if ($scope->getScenario()->hasTag('behat-steps-skip:bigPipeBeforeStep')) {
```

Comment:
BeforeStep does not have access to scenario so we communicate the scenario tag via property

Reply:
we already have a consistent way of skipping traits functionality using `'behat-steps-skip:' . __FUNCTION__`. Could you please elaborate why this mechanism is no good for skipping the resetting of the cookies on step.

I mean - you can put the property assignment under `'behat-steps-skip:' . __FUNCTION__` condition

Reply:
`if ($scope->getScenario()` Scope in BeforeStep does not have a method and does not have a way to access the Scenario